data_diff/cloud/data_source.py

```diff
@@ -0,0 +1,218 @@
+import enum
+import time
+from typing import List, Optional
+
+import pydantic
+import rich
+from rich.table import Table
+from rich.prompt import Confirm, Prompt, FloatPrompt, IntPrompt, InvalidResponse
+
+from .datafold_api import (
+    DatafoldAPI,
+    TCloudApiDataSourceConfigSchema,
+    TCloudApiDataSource,
+    TDsConfig,
+    TestDataSourceStatus,
+)
+
+
+DATA_SOURCE_TYPES_REQUIRED_SETTINGS = {
+    "bigquery": {"projectId", "jsonKeyFile", "location"},
+    "databricks": {"host", "http_password", "database", "http_path"},
+    "mysql": {"host", "user", "passwd", "db"},
+    "pg": {"host", "user", "port", "password", "dbname"},
+    "postgres_aurora": {"host", "user", "port", "password", "dbname"},
+    "postgres_aws_rds": {"host", "user", "port", "password", "dbname"},
+    "redshift": {"host", "user", "port", "password", "dbname"},
+    "snowflake": {"account", "user", "password", "warehouse", "role", "default_db"},
+}
```

Comment:
Good question!

I see only one place where pydantic validation might be used. Because we get possible database parameters from our REST API, i think only there we can conduct that validation in [DatafoldAPI.get_data_source_schema_config](https://github.com/datafold/data-diff/pull/466/files#diff-740a6ff327affccb60f289de72bf5d81ed37e2be3fb2f8702df577f35480e98cR175). Do you think it is a good idea?